src/status_im/chat/models/input.cljs

```diff
@@ -198,3 +199,24 @@
                         [(keyword (get-in params [i :name])) value]))
          (remove #(nil? (first %)))
          (into {}))))
+
+(defn- button-spamming? [last-sent-at cooldowns]
+  (< (- (datetime/timestamp) last-sent-at)
+     (+ const/spam-interval-ms (* 1000 cooldowns))))
+
+(defn- start-cooldown [{:keys [db]} cooldowns]
+  {:dispatch-later [{:dispatch [:enable-send-button]
+                     :ms       (const/cooldown-periods-ms cooldowns
+                                                          const/default-cooldown-period-ms)}]
+   :db             (assoc db
+                          :chat/cooldowns (if (= const/cooldown-reset-threshold cooldowns)
+                                            0
+                                            cooldowns)
+                          :chat/send-button-disabled? true)})
+
+(defn process-cooldown [{{:keys [chat/last-outgoing-message-sent-at chat/cooldowns current-chat-id] :as db} :db}]
+  (when (get-in db [:chats current-chat-id :group-chat])
```

Comment:
You could reuse:

https://github.com/status-im/status-react/blob/7ef3a5a5dab34500608ef30f3e2dd4b9ce95f55a/src/status_im/chat/models.cljs#L12

`(models.chat/multi-user-chat? current-chat-id cofx)`

hopefully there's no circular dependency :)